File: benchmark/apps/laravel-micro/database/seeders/WorldSeeder.php
<?php

namespace Database\Seeders;

use Illuminate\Database\Seeder;
use Illuminate\Support\Facades\DB;

class WorldSeeder extends Seeder
{
    public function run(): void
    {
        $rows = [];
        for ($i = 0; $i < 10000; $i++) {
            $rows[] = ['randomNumber' => random_int(1, 10000)];
        }
        DB::table('world')->insert($rows);
    }
}

File: benchmark/apps/ferro-micro/src/seeders/world_seeder.rs
//! World table seeder — inserts 10 000 rows with random numbers

use ferro::{async_trait, FrameworkError, Seeder};
use sea_orm::{DatabaseConnection, EntityTrait, Set};

use crate::models::world;

#[derive(Default)]
pub struct WorldSeeder;

#[async_trait]
impl Seeder for WorldSeeder {
    async fn run(&self, db: &DatabaseConnection) -> Result<(), FrameworkError> {
        use rand::Rng;
        // Collect all random numbers synchronously before any await point.
        // ThreadRng is !Send so it must not be held across .await.
        let rows: Vec<world::ActiveModel> = {
            let mut rng = rand::thread_rng();
            (1..=10_000)
                .map(|_| world::ActiveModel {
                    random_number: Set(rng.gen_range(1..=10_000)),
                    ..Default::default()
                })
                .collect()
        };
        world::Entity::insert_many(rows)
            .exec(db)
            .await
            .map_err(|e| FrameworkError::database(e.to_string()))?;
        Ok(())
    }
}
